logprep/connector/elasticsearch/output.py

```diff
@@ -280,7 +302,89 @@ def _handle_serialization_error(self, error: search.SerializationError):
         """
         raise FatalOutputError(self, f"{error.args[1]} in document {error.args[0]}")
 
-    def store(self, document: dict) -> bool:
+    def _is_message_exceeds_max_size_error(self, error):
+        if error.status_code == 429:
+            if error.error == "circuit_breaking_exception":
+                return True
+
+            if error.error == "rejected_execution_exception":
+                reason = error.info.get("error", {}).get("reason", {})
+                match = self.SIZE_ERROR_PATTERN.match(reason)
+                if match and int(match.group("size")) >= int(match.group("max_size")):
+                    return True
+        return False
+
+    def _handle_transport_error(self, error: search.exceptions.TransportError):
+        """Handle transport error for elasticsearch bulk indexing.
+
+        Discard messages that exceed the maximum size if they caused an error.
+
+        Parameters
+        ----------
+        error : TransportError
+           TransportError for the error message.
+
+        """
+        if self._max_message_size_byte is None:
+            raise error
+
+        if self._is_message_exceeds_max_size_error(error):
+            messages_under_size_limit = []
+            messages_over_size_limit = []
+            total_size = 0
+            for message in self._message_backlog:
+                message_size = get_json_dict_size_in_byte(message)
```

Comment:
I don't think this can be done, since one single field could be very large, which would then already break Logprep.

Reply:
Yeah that is true. I was thinking in the lines of serializing the whole dict into a string and then taking the size of the string. But I also really don't know if it would be such a performance boost. I was just thinking that recursively checking each element of maybe 10k messages could be quite costly.

Reply:
I did some crude tests to see what might be better.  If there are only small dicts, then msgspec is about 10x faster than json and the method, but the file size difference is signifikant. If the messages include some large ones, then the size difference becomes insignificant, but the method outperforms serializing it with msgspec. Since this check is only executed if there are large messages, I think it is ok to keep it as it is for now.

Reply:
Sorry, I did not see those comments. I've changed it now.